### ISSUE TITLE
Default value for boolean fields set as String

### ISSUE THREAD

**Issue by eikaramba** [directus/directus#14356]
### Describe the Bug

i have a boolean input field where i set the default value to false
![grafik](https://user-images.githubusercontent.com/1320525/178161266-1a7f3a1a-bb60-4074-be62-c5633eea2df0.png)

then i create a new entry without the boolean input (which should be set to the default value false)
![grafik](https://user-images.githubusercontent.com/1320525/178161289-1df27743-bba5-4b91-98a6-33871143f415.png)

However the field is created as a string "false" not false
![grafik](https://user-images.githubusercontent.com/1320525/178161307-2c9ff8a4-3ed1-4e6e-8311-f3d4e000757a.png)

### To Reproduce

1) create toggle field with default value false
2) post new collection entry without the toggle field
3) see response object

### What version of Directus are you using?

9.14.1

### What version of Node.js are you using?

16

### What database are you using?

SQLITE

### What browser are you using?

FF

### How are you deploying Directus?

local

**Comment by eikaramba** [comment]
interestingly this is what i see in the database. all other values with 0/1 are displayed correctly and are fine in the directus app as well. only the new test field with a value of false is also not correctly shown in the directus app (e.g. it has a checkmark in the overview)

**Comment by eikaramba** [comment]
Setting the default value to 0 manually in the sqlite database schema fixes everything. so i would argue that the default value set by directus is wrong

**Comment by azrikahar** [comment]
Initially I wasn't able to reproduce this:

https://user-images.githubusercontent.com/42867097/178181262-53de27c9-96c8-4a49-9c9a-2ac6766fba8b.mp4

But after doing further testings, it seems like it exclusively happens _AFTER_ updating the default value to `false`, but not when we initially set it as `false`. Here's another clip to show this clearer:

https://user-images.githubusercontent.com/42867097/178181385-5f53649b-cf54-40e2-be64-aad8be636f12.mp4

**Comment by aidenfoxx** [comment]
@azrikahar Feels like it could be similar to the issue addressed here https://github.com/knex/knex-schema-inspector/pull/117. When updating a column to nullable in most DB vendors, it tends to return string null. We should verify boolean cases for all DB vendors.

**Comment by azrikahar** [comment]
> @azrikahar Feels like it could be similar to the issue addressed here [knex/knex-schema-inspector#117](https://github.com/knex/knex-schema-inspector/pull/117). When updating a column to nullable in most DB vendors, it tends to return string null as the default. We should verify boolean cases for all DB vendors.

That's a good point. I haven't dig into this too much, but would this portion of the code already be the verification you're mentioning? 🤔 (which makes this even more odd on what regression could be happening, or was this never caught in the first place)

https://github.com/directus/directus/blob/9e387fe4b08b420fa1821a833c61db48c9926b65/api/src/utils/get-default-value.ts#L32-L42

**Comment by aidenfoxx** [comment]
@azrikahar I didn't realize there was already explicit boolean handling in that helper, but yes, that should handle this case. It could be that the helper doesn't account for whitespace? I know it was possible for nullable values to be returned as 'null ' from the schema query.

Either way, I personally don't like that this is handled in a helper function in Directus. It's my opinion that `knex-schema-inspector` should handle this as part of it's `parseDefaultValue` function. But we can worry about that later!

**Comment by azrikahar** [comment]
Looked into this a bit more, and I'm not sure if this is accurate, but here are my thoughts:

The API does seem to correctly pass `false` (boolean) as default value here in the last else statement:

https://github.com/directus/directus/blob/a803331b4aeaa5c027fe5e847129ad005cbba818/api/src/services/fields.ts#L586-L602

But it seems like the `defaultTo` in Knex sqlite `alterColumn()` here is not formatted correctly: https://github.com/knex/knex/blob/2dadde4214d9ee333adccfa517089647e94d23be/lib/dialects/sqlite3/schema/ddl.js#L102-L109

which may explain why this default value bug only happens when updating it, not when creating it.

Not sure if the formatDefault here was not ran somewhere _before_ alterColumn(): https://github.com/knex/knex/blob/2dadde4214d9ee333adccfa517089647e94d23be/lib/formatter/formatterUtils.js#L21-L36, or it's not formatting properly.

Would love to hear from @nickrum since I'm still not entirely sure here.

**Comment by nickrum** [comment]
Thanks for taking a look @azrikahar. `formatDefault()` is called over [here](https://github.com/knex/knex/blob/2dadde4214d9ee333adccfa517089647e94d23be/lib/dialects/sqlite3/schema/sqlite-tablecompiler.js#L54) before being passed to `SQLite3_DDL`. But it looks like it is called with the wrong type name. SQL uses `boolean` as the type name, while Knex internally uses `bool`.

Changing that line from

```js
formatDefault(col.modified['defaultTo'][0], type, this.client)
```

to

```js
formatDefault(col.modified['defaultTo'][0], col.type, this.client)
```

fixed the issue for me.

**Comment by nickrum** [comment]
This will be fixed by knex/knex#5319.

**Comment by rijkvanzanten** [comment]
Linear: [ENG-231](https://linear.app/directus/issue/ENG-231/default-value-for-boolean-fields-set-as-string)
